fx-src/FirefoxStorage.js

```diff
@@ -0,0 +1,213 @@
+          let schema = yield (connection.getSchemaVersion());
+
+          if (schema == 0) {
+
+            for (let statementName of createStatements) {
+              yield (connection.execute(statements[statementName]));
+            }
+
+            yield (connection.setSchemaVersion(1));
+          } else if (schema != 1) {
+            throw new Error("Unknown database schema: " + schema);
+          }
+        });
+        self.initialized = true;
+      } finally {
+        if (connection) {
+          connection.close();
+        }
+      }
+    });
+  }
+
+  _doOpenConnection() {
+    const opts = { path: "kinto.sqlite", sharedMemoryCache: false }
+    return Sqlite.openConnection(opts);
+  }
+
+  _doTransaction(statement, params){
+    const self = this;
+    return Task.spawn(function* (){
+      // get a connection
+      let connection = yield self._doOpenConnection();
+
+      // ensure we're initialized
+      yield self.init;
+
+      //execute the transaction
+      try {
+        return yield connection.executeCached(statement, params);
+      }
+      finally {
+        // ensure the connection is closed
+        connection.close();
+      }
+    });
+  }
+
+  clear() {
+    const params = {collection_name: this.dbname};
+    return this._doTransaction(statements["clearData"], params);
+  }
+
+  create(record) {
+    const params = {
+      collection_name: this.dbname,
+      record_id: record.id,
+      record: JSON.stringify(record)
+    };
+    return this._doTransaction(statements["createData"], params).then(() => {
+      return record;
+    });
+  }
+
+  update(record) {
+    const params = {
+      collection_name: this.dbname,
+      record_id: record.id,
+      record: JSON.stringify(record)
+    };
+    return this._doTransaction(statements["updateData"], params).then(() => {
+      return record;
+    });
+  }
+
+  get(id) {
+      const params = {
+        collection_name: this.dbname,
+        record_id: id,
+      };
+      return this._doTransaction(statements["getRecord"], params).then(result => {
+        if (result.length == 0) {
+          return;
+        }
+        return JSON.parse(result[0].getResultByName("record"));
+
+      });
+    }
+
+  delete(id) {
+    const params = {
+      collection_name: this.dbname,
+      record_id: id,
+    };
+    return this._doTransaction(statements["deleteData"], params);
+  }
+
+  list() {
+    const params = {
+      collection_name: this.dbname,
+    };
+    return this._doTransaction(statements["listRecords"], params).then(result => {
+      const records = [];
+      for (let k = 0; k < result.length; k++) {
+        let row = result[k];
+        records.push(JSON.parse(row.getResultByName("record")));
+      }
+      return records;
+    });
+  }
+
+  saveLastModified(lastModified) {
+    const params = {
+      collection_name: this.dbname,
+      last_modified: lastModified,
+    };
+    return this._doTransaction(statements["saveLastModified"], params);
```

Comment:
The Collection API expects `saveLastModified` to resolve with saved `lastModified` value, as an int or `null`:

```js
var value = parseInt(lastModified, 10) || null;
```